modules/img2img.py

```diff
@@ -46,7 +46,10 @@ def process_batch(p, input_dir, output_dir, inpaint_mask_dir, args):
         if state.interrupted:
             break
 
-        img = Image.open(image)
+        try:
+            img = Image.open(image)
+        except UnidentifiedImageError:
+            continue

```

Comment:
Please put in a global function so that scripts and extensions can use it too.

Reply:
Particularly, [controlnet batch](https://github.com/Mikubill/sd-webui-controlnet/pull/683) needs to list valid images from a batch dir.

Reply:
Do you think it makes sense to create a wrapper for a library call to Pillow's PIL.Image.open() ? It's already handling missing files and non-images as exceptions.

I would tend to think it should be the caller's responsibility to either making sure it's called with proper parameters (i.e. an existing file, or a proper image file) or catching any potential exception if they don't.

Depending on where it's called, some may want to do different things based on the context or which exceptions could be thrown. In this case here, we can skip the file. But it some other context we might want to do something else.

I see that in some places where Image.open() is used, the files are already checked for existence `(generation_parameters_copy_paste.image_from_url_text()`), or exceptions are being handled (`images.image_data`, the second use inside `postprocessing.run_postprocessing()`)

I see only two places where it's not necessarily handled (`ui.process_interrogate()` and the first occurrence inside `postprocessing.run_postprocessing()`) but I would tend to think these should be fixed there according to the context.

I'd rather fix these last two as well in the same PR and let each other caller fix their calls if needed (i.e. each extension takes care of input validation and/or calling properly and/or catching exceptions, since all is already handled by exceptions in `PIL.Image.open()`).

Reply:
Right, but ultimately, let's say we write a wrapper to open a file. Then it would return None if it's not a file.

- Caller would have to test for None as possible return anyway.
- Caller wouldn't know if it's because the file doesn't exist or if the format of the file is invalid.

I imagine you're talking about having a specific procedure to retrieve the images just for the case of using a source input directory (for instance get_images_from_directory(path_to_dir) that would return all the images that PIL.Images can open in that folder).

But the problem is that in some places the input is provided as a list of paths, sometimes as a name of a dir, sometimes as a dict of files, so this means refactoring some areas.

Reply:
To be honest I have not looked at the context maybe what I'm asking for is out of scope for this PR. I was just going through the ones open in the repo and noticed this one was related to what I was working on, so I wanted to make the webui exposed code more user friendly and reusable. Feel free to close this discussion if that would be too much work, we can look into this in a followup PR maybe.

Reply:
Right maybe that would be best handled in a separate PR, as the scope is a bit more involved.